app/src/main/java/com/codepath/apps/restclienttemplate/fragment/ComposeFragment.java

```diff
@@ -0,0 +1,152 @@
+import com.codepath.apps.restclienttemplate.TwitterApp;
+import com.codepath.apps.restclienttemplate.TwitterClient;
+import com.codepath.apps.restclienttemplate.activities.TimelineActivity;
+import com.codepath.apps.restclienttemplate.models.Tweet;
+import com.codepath.asynchttpclient.callback.JsonHttpResponseHandler;
+
+import org.json.JSONException;
+import org.parceler.Parcels;
+
+import okhttp3.Headers;
+
+public class ComposeFragment extends DialogFragment implements TextView.OnEditorActionListener {
+
+    private static final String TAG = "ComposeFragment";
+    private static final String TARGET_USER_KEY = "TargetUser";
+    private static int MAX_TWEET_LENGTH = 280;
+    private EditText etCompose;
+    private Button btTweet;
+    private String tweetContent;
+    private TwitterClient client;
+    private String targetUser;
+    private ImageView ivClose;
+    private Tweet returnTweet;
+
+    public ComposeFragment() {
+    }
+
+    @Override
+    public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
+        if (EditorInfo.IME_ACTION_DONE == i) {
+            // Return input text back to activity through the implemented listener
+            ComposeFragmentListener listener = (ComposeFragmentListener) getActivity();
+            listener.onFinishComposeTweet(returnTweet);
+            // Close the dialog and return back to the parent activity
+            dismiss();
+            return true;
+        }
+        return false;    }
+
+    public interface ComposeFragmentListener {
+        void onFinishComposeTweet(Tweet returnTweet);
+    }
+
+    // Replace normal constructor
+    public static ComposeFragment newInstance(String targetUser) {
+        ComposeFragment frag = new ComposeFragment();
+        Bundle args = new Bundle();
+        args.putString(TARGET_USER_KEY, targetUser);
+        frag.setArguments(args);
+        return frag;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_compose, container);
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // Get field from view
+        etCompose = (EditText) view.findViewById(R.id.etCompose);
+        btTweet = (Button) view.findViewById(R.id.btnTweet);
+        client = TwitterApp.getRestClient(getActivity());
+        ivClose = (ImageView) view.findViewById(R.id.ivClose);
+
+        // Fetch arguments from bundle and set title
+        targetUser = getArguments().getString(TARGET_USER_KEY);
+        if (targetUser != "") {
+            etCompose.setText("@" + targetUser + " ");
+            btTweet.setText("Reply");
+        }
+
+        etCompose.requestFocus();
+        btTweet.setOnClickListener(publish);
+        ivClose.setOnClickListener(close);
```

Comment:
If a user closes the compose fragment by pressing the "x" button, they might want to just dismiss the window and go back to where they were in the tweets instead of going to the top of all the tweets